db/crud.go

```diff
@@ -1106,13 +1106,25 @@ func (db *DatabaseCollectionWithUser) PutExistingCurrentVersion(ctx context.Cont
 			if addNewerVersionsErr != nil {
 				return nil, nil, false, nil, addNewerVersionsErr
 			}
+			if docHLV.MergeVersions != nil {
+				if doc.HLV.MergeVersions == nil {
+					doc.HLV.MergeVersions = make(map[string]uint64)
+				}
+				doc.HLV.MergeVersions = docHLV.MergeVersions
+			}
 		} else {
 			if !docHLV.IsInConflict(*doc.HLV) {
 				// update hlv for all newer incoming source version pairs
 				addNewerVersionsErr := doc.HLV.AddNewerVersions(docHLV)
 				if addNewerVersionsErr != nil {
 					return nil, nil, false, nil, addNewerVersionsErr
 				}
+				if docHLV.MergeVersions != nil {
+					if doc.HLV.MergeVersions == nil {
+						doc.HLV.MergeVersions = make(map[string]uint64)
+					}
+					doc.HLV.MergeVersions = docHLV.MergeVersions
+				}
```

Comment:
Same code in _both_ the if and else case seems like it's in the wrong place.

Why was this nessesary, and could it not be outside of the if/else if we always do the same thing.